tests/functional/states/test_kubernetes.py

```diff
@@ -363,37 +363,33 @@ def test_deployment_present(kubernetes, deployment, testmode, kubernetes_exe):
     assert ret.result in (None, True)
     assert (ret.result is None) is testmode
     if not testmode:
-        assert ret.changes["spec"]["replicas"] == 2
+        assert ret.changes["new"]["spec"]["replicas"] == 2
         deployment_state = kubernetes_exe.show_deployment(
             name=deployment["name"], namespace=deployment["namespace"]
         )
         assert deployment_state["metadata"]["name"] == deployment["name"]
         assert deployment_state["spec"]["replicas"] == 2
     else:
-        # FIXME: Changes should be reported in test mode
-        assert not ret.changes
+        assert ret.changes["new"]["spec"]["replicas"] == 2
         deployment_state = kubernetes_exe.show_deployment(
             name=deployment["name"], namespace=deployment["namespace"]
         )
         assert deployment_state is None
 
-    # Comment out idempotent test for now
-    # TODO: The state module needs fixed to handle proper present functionality
-
 
-# def test_deployment_present_idempotency(kubernetes, deployment):
-#     """
-#     Test kubernetes.deployment_present is idempotent
-#     """
-#     ret = kubernetes.deployment_present(
-#         name=deployment["name"],
-#         namespace=deployment["namespace"],
-#         spec=deployment["spec"],
-#         wait=True,
-#     )
-#     assert ret.result is True
-#     assert "already exists" in ret.comment
-#     assert not ret.changes
+def test_deployment_present_idempotency(kubernetes, deployment):
+    """
+    Test kubernetes.deployment_present is idempotent
+    """
+    ret = kubernetes.deployment_present(
+        name=deployment["name"],
+        namespace=deployment["namespace"],
+        spec=deployment["spec"],
+        wait=True,
+    )
```

Comment:
nit: I'd also test with `test=true` for completeness' sake.

```suggestion
def test_deployment_present_idempotency(kubernetes, deployment, testmode):
    """
    Test kubernetes.deployment_present is idempotent
    """
    ret = kubernetes.deployment_present(
        name=deployment["name"],
        namespace=deployment["namespace"],
        spec=deployment["spec"],
        wait=True,
        test=testmode,
    )
```